thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dashboard/resources/RootResource.java

```diff
@@ -0,0 +1,201 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.pinot.thirdeye.dashboard.resources;
+
+import com.google.inject.Inject;
+import javax.ws.rs.Path;
+import org.apache.pinot.thirdeye.api.application.ApplicationResource;
+import org.apache.pinot.thirdeye.api.user.dashboard.UserDashboardResource;
+import org.apache.pinot.thirdeye.dashboard.resources.v2.AnomaliesResource;
+import org.apache.pinot.thirdeye.dashboard.resources.v2.ConfigResource;
+import org.apache.pinot.thirdeye.dashboard.resources.v2.DataResource;
+import org.apache.pinot.thirdeye.dashboard.resources.v2.DetectionAlertResource;
+import org.apache.pinot.thirdeye.dashboard.resources.v2.alerts.AlertResource;
+import org.apache.pinot.thirdeye.dataset.DatasetAutoOnboardResource;
+import org.apache.pinot.thirdeye.datasource.sql.resources.SqlDataSourceResource;
+import org.apache.pinot.thirdeye.detection.DetectionResource;
+import org.apache.pinot.thirdeye.detection.yaml.YamlResource;
+
+/**
+ * This is the Jersey root endpoint "/"
+ * All child endpoints should be registered here with method level @Path annotations.
+ *
+ */
+@Path("/")
+public class RootResource {
+
+  private final AlertResource alertResource;
+  private final AdminResource adminResource;
+  private final AnomaliesResource anomaliesResource;
+  private final ApplicationResource applicationResource;
+  private final AutoOnboardResource autoOnboardResource;
+  private final CacheResource cacheResource;
+  private final ConfigResource configResource;
+  private final CustomizedEventResource customizedEventResource;
+  private final DashboardResource dashboardResource;
+  private final DataResource dataResource;
+  private final DatasetAutoOnboardResource datasetAutoOnboardResource;
+  private final DetectionResource detectionResource;
+  private final DetectionAlertResource detectionAlertResource;
+  private final EntityMappingResource entityMappingResource;
+  private final OnboardDatasetMetricResource onboardDatasetMetricResource;
+  private final SqlDataSourceResource sqlDataSourceResource;
+  private final ThirdEyeResource thirdEyeResource;
+  private final UserDashboardResource userDashboardResource;
+  private final YamlResource yamlResource;
+
+  @Inject
+  public RootResource(
```

Comment:
Swagger isn't working if the resources are put under this class.

Reply:
Thanks for trying this out. Really appreciate it. I missed adding the @Api Tag to the RootResource. Fixed it.